src/tutorials/0006/05.md

```diff
@@ -0,0 +1,25 @@
+So now our CID in binary, looks like this:
+
+`<cid-version><multicodec><multihash>`
+
+But binary CIDs are not very human friendly, so just like we saw in the beginning of this tutorial, we can represent these binary CIDs in a string form (binary data represented as text). Example:
+
+`bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi`
+
+But how we do know which base encoding was the string form of the hash converted with?
+
+In `CIDv0` hashes are **always** encoded with `base58btc`. _Always_. This means that we can safely interpret `CIDv0` hashes assuming they are using `base58btc`. But, for performance reasons, we may want to support multiple base encodings. For that, you guessed it, we can add another prefix!
+
+## [Multibase](https://github.com/multiformats/multibase)
+
+![Multibase Prefix](tutorial-assets/T0006L05-multibase-prefix.jpg)
+
+A detail about this prefix, is that it is **only used in the string form of the CID**.
+
+So let's see two examples of CIDs in the string form:
+
+![String form examples](tutorial-assets/T0006L05-string-form.jpg)
+
+The first one is a `CIDv0` because it starts with `Qm...`. All Hashes that start with `Qm` can safely be interpreted as `base58btc` as a CID of version 0.
+
+The second one starts with `bafy...` and the first character `b` is the base encoding prefix indentifier because, usually, most implementations of IPFS by default encode the CIDs using `base32`, which its identifier character is `b`. For the complete list of identifiers of `multibase` see [this table](https://github.com/multiformats/multibase/blob/master/multibase.csv).
```

Comment:
There are some variations, but `bafy...` CIDs are identifiable as CIDv1.
CIDv2 would not start with `bafy...` because of the version prefix if I'm not mistaken.

Reply:
@zebateira actually it's a bit more involved than that. Consider the following set of examples ( the `b` is missing due to cyberchef not being awesome ):
1. [DAG-PB through sha256](https://gchq.github.io/CyberChef/#recipe=From_Hex('Auto')To_Base32('a-z2-7%3D')&input=XDAxIFw3MCBcMTIgXDIwIFxmZg) ( currently most prevalent )
2. [raw leaves through sha256](https://gchq.github.io/CyberChef/#recipe=From_Hex('Auto')To_Base32('a-z2-7%3D')&input=XDAxIFw1NSBcMTIgXDIwIFxmZg) ( will be default for all leaf nodes in ipfs 0.6 )
3. [DAG-CBOR through sha256](https://gchq.github.io/CyberChef/#recipe=From_Hex('Auto')To_Base32('a-z2-7%3D')&input=XDAxIFw3MSBcMTIgXDIwIFxmZg) ( definitely what we will switch to wholesale before ipfs 1.0 )

Same as above but with blake2b ( likely candidate for ipfs 1.0 )
4. [DAG-PB/blake](https://gchq.github.io/CyberChef/#recipe=From_Hex('Auto')To_Base32('a-z2-7%3D')&input=XDAxIFw3MCBcYTAgXGU0IFwwMiBcMjAgXGZm)
5. [raw/blake](https://gchq.github.io/CyberChef/#recipe=From_Hex('Auto')To_Base32('a-z2-7%3D')&input=XDAxIFw1NSBcYTAgXGU0IFwwMiBcMjAgXGZm)
6. [DAG-CBOR/blake](https://gchq.github.io/CyberChef/#recipe=From_Hex('Auto')To_Base32('a-z2-7%3D')&input=XDAxIFw3MSBcYTAgXGU0IFwwMiBcMjAgXGZm)

Then there is *inlining/identity-hashing* which has yet another set of worms. The only thing you can *somewhat* hint at is `baf` for "maybe a CIDv1", due to the fact that the most widely used codecs are `0x55`, `0x70` and `0x71` ( `bafk...`, `bafy...`, `bafy...` respectively )

But at the end of the day @terichadbourne is correct: you should really only emphasize `b` and that's that

Reply:
@ribasushi makes sense 👍 will update then!